src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -60,19 +60,32 @@ def completer(prefix, action, parsed_args, **kwargs):#pylint: disable=unused-arg
             return [r.name for r in get_resources_in_subscription(resource_type=resource_type)]
     return completer
 
-def get_enum_type_completion_list(enum_type=None):
-    ENUM_MAPPING_KEY = '_value2member_map_'
-    def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
-        return list(enum_type.__dict__[ENUM_MAPPING_KEY].keys())
-    return completer
-
 def get_generic_completion_list(generic_list):
     def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
         return generic_list
     return completer
 
-def get_enum_choices(enum_type):
-    return [x.value for x in enum_type]
+class CaseInsenstiveList(list): # pylint: disable=too-few-public-methods
```

Comment:
Also, since argcomplete uses choice lists as default completers, I verified that the tab completion behavior using these choice lists still works.

`az storage account create --sku prem[tab][tab]`

correctly yields
`az storage account create --sku Premium_LRS`